### PR TITLE
change indexer interface master

### PR DESCRIPTION
**Pull request by miiu96**
- Changed indexer interface in order to prepare the ground for the refactored indexer.
- Removed unused mocks from `core` package

### CODE REVIEW

File: node/mock/indexerMock.go [hunk 1]
```diff
@@ -34,12 +34,12 @@ func (im *IndexerMock) UpdateTPS(_ statistics.TPSBenchmark) {
 }
 
 // SaveRoundsInfo -
-func (im *IndexerMock) SaveRoundsInfo(_ []workItems.RoundInfo) {
+func (im *IndexerMock) SaveRoundsInfo(_ []*indexer.RoundInfo) {
 	panic("implement me")
 }
 
 // SaveValidatorsRating --
```
Comment (iulianpascalau) [inline node/mock/indexerMock.go:41]: maybe drop the second -?

File: node/mock/indexerMock.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ type IndexerMock struct {
 }
 
 // SaveBlock -
-func (im *IndexerMock) SaveBlock(_ data.BodyHandler, _ data.HeaderHandler, _ map[string]data.TransactionHandler, _ []uint64, _ []string, _ []byte) {
+func (im *IndexerMock) SaveBlock(_ *indexer.ArgsSaveBlockData) {
 	panic("implement me")
```
Comment (iulianpascalau) [inline node/mock/indexerMock.go:19]: maybe rename this struct to a stub and add those ...Called handlers?

File: process/block/metablock.go [hunk 3]
```diff
@@ -555,14 +556,10 @@ func (mp *metaProcessor) indexBlock(
...
-	txPool := mp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock)
-	scPool := mp.txCoordinator.GetAllCurrentUsedTxs(block.SmartContractResultBlock)
-
-	for hash, tx := range scPool {
-		txPool[hash] = tx
-	}
-	for hash, tx := range rewardsTxs {
-		txPool[hash] = tx
+	pool := &indexer.Pool{
+		Txs:     mp.txCoordinator.GetAllCurrentUsedTxs(block.TxBlock),
```
Comment (iulianpascalau) [inline process/block/metablock.go:560]: this works as expected because the transactionCoordinator.GetAllCurrentUsedTxs create new maps on each call.

**Review by iulianpascalau — CHANGES_REQUESTED** [review]
Good impl! :+1: will re-evaluate after the code freeze on 'master' branch will be lifted

File: node/mock/indexerMock.go (hunk 1 above)
Comment (miiu96) [reply node/mock/indexerMock.go:41]: done.

File: node/mock/indexerMock.go (hunk 2 above)
Comment (miiu96) [reply node/mock/indexerMock.go:19]: done.

File: process/block/metablock.go (hunk 3 above)
Comment (miiu96) [reply process/block/metablock.go:560]: Yes. This was used before and all the transactions were indexed.

Comment (iulianpascalau) [reply process/block/metablock.go:560]: Yup

**Review by iulianpascalau — APPROVED** [review]

File: cmd/storer2elastic/dataprocessor/dataReplayer_test.go [hunk 4]
```diff
@@ -263,6 +264,129 @@ func TestDataReplayer_Range_EpochStartMetaBlockFoundInNonceHashStorageShouldWork
 	_ = dr.Range(handlerFunc)
 }
 
+func TestDataReplayer_Range_TransactionInPool(t *testing.T) {
```
Comment (raduchis) [inline cmd/storer2elastic/dataprocessor/dataReplayer_test.go:267]: This test has no asserts. Some other tests with Range have no asserts, is it ok?

File: cmd/storer2elastic/mock/elasticIndexerStub.go [hunk 5]
```diff
@@ -1,22 +1,21 @@
...
 )
 
 // ElasticIndexerStub -
 type ElasticIndexerStub struct {
-	SetTxLogsProcessorCalled func(txLogsProc process.TransactionLogProcessorDatabase)
-	SaveBlockCalled          func(body data.BodyHandler, header data.HeaderHandler,
-		txPool map[string]data.TransactionHandler, signersIndexes []uint64, notarizedHeadersHashes []string, headerHash []byte)
-	SaveRoundsInfosCalled       func(roundsInfos []workItems.RoundInfo)
+	SetTxLogsProcessorCalled    func(txLogsProc process.TransactionLogProcessorDatabase)
+	SaveBlockCalled             func(args *indexer.ArgsSaveBlockData)
```
Comment (raduchis) [inline cmd/storer2elastic/mock/elasticIndexerStub.go:14]: :+1:

File: process/block/metablock.go (hunk 3 above)
Comment (raduchis) [reply process/block/metablock.go:560]: no need to add type invalid and receipts?

Comment (miiu96) [reply process/block/metablock.go:560]: On the metachain we don't have receipts and invalid transactions. @sasurobert can you confirm this, please.

File: go.mod [hunk 6]
```diff
@@ -5,7 +5,7 @@ go 1.13
 require (
 	github.com/ElrondNetwork/arwen-wasm-vm v1.1.1
 	github.com/ElrondNetwork/concurrent-map v0.1.3
-	github.com/ElrondNetwork/elastic-indexer-go v1.0.2-0.20210310091712-487aeb322b48
+	github.com/ElrondNetwork/elastic-indexer-go v1.0.2-0.20210319083359-c0367f772e17
```
Comment (iulianpascalau) [inline go.mod:8]: why not a release?

Comment (miiu96) [reply go.mod]: done.

**Review by iulianpascalau — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Review by raduchis — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by raduchis — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System Test passed

**Pull request merged**